app/src/main/java/com/spencerpages/MainActivity.java

```diff
@@ -444,90 +462,80 @@ public void onClick(DialogInterface dialog, int item) {
 			// Shared with PC so can't write to it
     		showCancelableAlert(mRes.getString(R.string.cannot_rd_ext_media_shared));
-        	return;		
+        	return;
 		} else {
 			// Doesn't exist, so notify user
 			showCancelableAlert(mRes.getString(R.string.cannot_rd_ext_media_state, state));
         	return;
 		}
-		
+
     	//http://stackoverflow.com/questions/3551821/android-write-to-sd-card-folder
     	File sdCard = Environment.getExternalStorageDirectory();
-
     	String path = sdCard.getAbsolutePath() + EXPORT_FOLDER_NAME;
     	File dir = new File(path);
 
     	if(!dir.isDirectory()){
-    		// The directory doesn't exist, notify the user
-    		showCancelableAlert(mRes.getString(R.string.cannot_find_export_dir, path));
-        	return;
-    	}
-    	
-    	File file;
-    	FileInputStream f;
-    	InputStreamReader in_;
-    	BufferedReader in;
+            // The directory doesn't exist, notify the user
+            showCancelableAlert(mRes.getString(R.string.cannot_find_export_dir, path));
+            return;
+        }
+
     	boolean errorOccurred = false;
 
         // Read the database version
-        file = new File(dir, "database_version.txt");
+        File inputFile = new File(dir, "database_version.txt");
+        BufferedReader in = openFileForReading(inputFile);
+        if(in == null) return;
 
         try {
-            f = new FileInputStream(file);
-            in_ = new InputStreamReader(f, "UTF8");
-            in = new BufferedReader(in_);
-            this.mDatabaseVersion = Integer.parseInt(in.readLine());
-
+            mDatabaseVersion = Integer.parseInt(in.readLine());
         } catch (Exception e) {
-            showCancelableAlert(mRes.getString(R.string.error_open_file_reading, file.getAbsolutePath()));
+            showCancelableAlert(mRes.getString(R.string.error_reading_file, inputFile.getAbsolutePath()));
             return;
         }
 
-        if(-1 == this.mDatabaseVersion){
+        if(-1 == mDatabaseVersion){
             showCancelableAlert(mRes.getString(R.string.error_db_version));
             return;
         }
 
-        try {
-            in.close();
-        } catch (IOException e) {
-            showCancelableAlert(mRes.getString(R.string.error_closing_input_file, file.getAbsolutePath()));
+        if(closeInputFile(in, inputFile)){
             return;
         }
 
         // Read the collection_info table
-    	file = new File(dir, "list-of-collections.csv");
+        inputFile = new File(dir, "list-of-collections.csv");
+        in = openFileForReading(inputFile);
+        if(in == null) return;
 
-    	try {
-			f = new FileInputStream(file);
-			in_ = new InputStreamReader(f, "UTF8");
-			in = new BufferedReader(in_);
-		} catch (Exception e) {
-			showCancelableAlert(mRes.getString(R.string.error_open_file_reading, file.getAbsolutePath()));
-			return;
-		}
-    	
 		ArrayList<String[]> collectionInfo = new ArrayList<>();
+        String line;
```

Comment:
Should the line declaration be moved back into the 'try' block to limit its scope?